cpp/src/arrow/util/launder.h

```diff
@@ -22,7 +22,7 @@
 namespace arrow {
 namespace internal {
 
-#if __cplusplus >= 201703L
+#ifdef __cpp_lib_launder
```

Comment:
`__cpp_lib_launder` is not guaranteed to always be define, so checking C++ version is a more portable way in this case.

Reply:
This could be potentially be tested in cross platform way using [newer CMake](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html), but I am not sure if this particular one is added to CMake. Have to check.

Reply:
if __cpp_lib_launder defined, std::launder is available for sure, otherwise, it could be not present even for specified __cplusplus version.
in my opinion, best solution here macro option like HAS_STD_LAUNDER defined at cmake configuration stage, but proposed variant should work too